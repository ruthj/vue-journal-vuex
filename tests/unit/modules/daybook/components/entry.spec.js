import { shallowMount } from '@vue/test-utils'
import Entry from '../../../../../src/modules/daybook/components/Entry.vue'
import {journalState} from '../../../mock-data/test-journal-state'

describe('Entry component', () => {

    const mockRouter ={
        push: jest.fn()
    }

    let wrapper
    
    beforeEach(() => {
        wrapper = shallowMount(Entry, {
            props: {
                entry: journalState.entries[0]
            },
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })
    })

    test('should match with snapshot', () => {
        expect (wrapper.html()).toMatchSnapshot()
    })

    test('should redirect when click entry', () => {
       
        const clickEvent = wrapper.find('.entry-container')//finds by class
        clickEvent.trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
    })

    test('computed properties test', () => {
       
        const dates = wrapper.vm
        expect(dates.day).toBeTruthy()
        expect(dates.month).toBeTruthy()
        expect(dates.yearDay).toBeTruthy()
        //console.log(wrapper.vm)

    })

})