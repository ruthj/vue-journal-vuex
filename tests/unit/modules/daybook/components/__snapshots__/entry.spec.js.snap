// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component should match with snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">30</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2021, Martes</span></div>
  <div class="entry-description">MOCK DATA</div>
</div>
`;
